mods/ra/maps/soviet-02a/soviet02a.lua

```diff
@@ -0,0 +1,170 @@
+
+
+WorldLoaded = function()
+	player = Player.GetPlayer("USSR")
+	germany = Player.GetPlayer("Germany")
+
+	Trigger.OnObjectiveAdded(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+	Trigger.OnObjectiveCompleted(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+	Trigger.OnObjectiveFailed(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+	end)
+
+	CommandCenterIntact = player.AddPrimaryObjective("Protect the Command Center.")
+	DestroyallAllied = player.AddPrimaryObjective("Destroy all Allied units and structures.")
+
+	Trigger.OnPlayerWon(player, function()
+		Media.PlaySpeechNotification(player, "MissionAccomplished")
+	end)
+	Trigger.OnPlayerLost(player, function()
+		Media.PlaySpeechNotification(player, "MissionFailed")
+	end)
+
+	Camera.Position	= CameraWaypoint.CenterPosition
+
+	Trigger.OnKilled(CommandCenter, function()
+		player.MarkFailedObjective(CommandCenterIntact)
+	end)
+
+	Trigger.AfterDelay(0, function()
+		local buildings = Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(self) return self.Owner == germany and self.HasProperty("StartBuildingRepairs") end)
+		Utils.Do(buildings, function(actor)
+			Trigger.OnDamaged(actor, function(building, attacker)
+				if building.Owner == germany and building.Health < building.MaxHealth * 0.8 then
+					building.StartBuildingRepairs()
+					if attacker.Type ~= "Yak" and not AlreadyHunting then
+						AlreadyHunting = true
+						Utils.Do(germany.GetGroundAttackers(), function(unit)
+							Trigger.OnIdle(unit, unit.Hunt)
+						end)
+					end
+				end
+			end)
+		end)
+
+		-- Find the bridge actors
+		bridgepart1 = Map.ActorsInBox(waypoint23.CenterPosition, waypoint49.CenterPosition, function(self) return self.Type == "br1" end)[1]
+		bridgepart2 = Map.ActorsInBox(waypoint23.CenterPosition, waypoint49.CenterPosition, function(self) return self.Type == "br2" end)[1]
+	end)
+
+	-- Discover the area around the bridge exposing the two german soldiers
+	-- When the two infantry near the bridge are discovered move them accross the bridge to waypoint4
+	-- in the meanwhile one USSR soldier hunts them down
+	Trigger.AfterDelay(DateTime.Seconds(1), function()
+		Actor.Create("camera", true, { Owner = player, Location = waypoint23.Location })
+
+		Utils.Do(FleeingUnits, function(unit)
+			unit.Move(waypoint4.Location)
+		end)
+		Follower.AttackMove(waypoint4.Location)
+	end)
+
+	-- To make it look more smooth we will blow up the bridge when the barrel closest to it blows up
+	Trigger.OnAnyKilled({ BridgeBarrel1, BridgeBarrel2 }, function()
+		-- Destroy the bridge
+		if not bridgepart1.IsDead then
+			bridgepart1.Kill()
+		end
+		if not bridgepart2.IsDead then
+			bridgepart2.Kill()
+		end
+
+		-- Move units towards the oilpump
+		--[[Utils.Do(FleeingUnits, function(unit)
```

Comment:
I commented this out as I don't get the reason for this.
What should I do with it?